### ISSUE TITLE
mace output and tensorflow output are different after ArgMax

### ISSUE THREAD

**Issue by FrisaSZ** [XiaoMi/mace#426]
### System information
**I use docker image**
**repo**: registry.cn-hangzhou.aliyuncs.com/xiaomimace/mace-dev-lite
**tag**: latest
**image id**: 7a74a198a246
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04.5 LTS
- **NDK version(e.g., 15c)**: r15c
- **GCC version(if compiling for host, e.g., 5.4.0)**: 5.4.0
- **MACE version (Use the command: git describe --long --tags)**: v0.10.0-192-gabbc3d4
- **Python version(2.7)**: 2.7.14
- **Bazel version (e.g., 0.13.0)**: 0.16.0
- **tensorflow version**: 1.12.0

#### 1. use last layer (argmax) as output
```yaml
library_name: argmaxtest
target_abis: [host]
model_graph_format: file
model_data_format: file
models:
  argmaxtest:
    platform: tensorflow
    model_file_path: /home/houjx/fsz_work/rtseg/out_dir/model.pb
    model_sha256_checksum: d1f7e4f8e3cced66272f046b406e89c5e5f69e10166a0655919c1cdca7742916
    subgraphs:
      - input_tensors: in_node
        output_tensors: out_node
        input_shapes: 1,256,256,3
        output_shapes: 1,256,256
    runtime: cpu
    limit_opencl_kernel_time: 0
    nnlib_graph_mode: 0
    obfuscate: 0
    winograd: 0
```

#### 2. use second last layer (logits) as output
```yaml
library_name: argmaxtest
target_abis: [host]
model_graph_format: file
model_data_format: file
models:
  argmaxtest:
    platform: tensorflow
    model_file_path: /home/houjx/fsz_work/rtseg/out_dir/model.pb
    model_sha256_checksum: d1f7e4f8e3cced66272f046b406e89c5e5f69e10166a0655919c1cdca7742916
    subgraphs:
      - input_tensors: in_node
        output_tensors: logits/logits
        input_shapes: 1,256,256,3
        output_shapes: 1,256,256,4
    runtime: cpu
    limit_opencl_kernel_time: 0
    nnlib_graph_mode: 0
    obfuscate: 0
    winograd: 0
```

### Describe the problem
When I use the second last layer (logits) as the output node, the mace outputs and tensorflow outputs are very close. Validation results are:
```bash
* Validate with tensorflow
logits/logits MACE VS TENSORFLOW similarity: 0.9999999999999207 , sqnr: 6217423659917.2295 , pixel_accuracy: 1.0
******************************************
          Similarity Test Passed          
******************************************

Validation done!
```
If I use the last layer (argmax) as the output node, the mace outputs are most zeros. Therefore, validate failed:
```bash
* Validate with tensorflow
out_node MACE VS TENSORFLOW similarity: 0.1046430195741308 , sqnr: 1.0 , pixel_accuracy: 0.93359375
ERROR: [] /home/houjx/fsz_work/mace/tools/validate.py:113: ******************************************
          Similarity Test Failed          
******************************************
```

### To Reproduce

1. cd /path/to/mace
2. download **model.pb** from [here](https://drive.google.com/open?id=1JG0UNQyUae2XhlCPaOwf-9PSdvcITR-W). (or you can run the code [here](https://gist.github.com/FrisaSZ/b24c317e2c3888b5430770c94fd9c944) to generate a pb file)
3. python tools/converter.py convert --config_file=those/two/yml/files/above
4. python tools/converter.py run --config_file=those/two/yml/files/above --validate

**Comment by TonyMou** [comment]
@FrisaSZ  MACE does not support different types in argmax node. Suggest you set the output_type of ArgMax node to float.

**Comment by liyinhgqw** [comment]
Please set 
```
input_data_formats: None
output_data_formats: None
```

and see how it goes ?

**Comment by liyinhgqw** [comment]
It has something to do with data format. We are working on it. Use logits for temporary workaround please.

**Comment by FrisaSZ** [comment]
@liyinhgqw Thanks for the reply. 
I have used logits as the output node and implemented argmax function myself. But still looking forward to the solution in mace.

**Comment by yejw5** [comment]
@FrisaSZ Test the lastest master branch?

**Comment by FrisaSZ** [comment]
@yejw5 Hi. Now the similarity test passed. Thanks.
